src/intervals/intervals.jl

```diff
@@ -18,12 +18,18 @@ immutable Interval{T<:Real} <: AbstractInterval
         end
 
         if a > b
-            (isinf(a) && isinf(b)) && return new(a, b)  # empty interval = [∞,-∞]
-
+            # empty interval = [∞,-∞]
+            (isinf(a) && isinf(b)) && return new(a, b)
             throw(ArgumentError("Must have a ≤ b to construct Interval(a, b)."))
         end
 
-
+        # The IEEE Std 1788-2015 states that a < Inf and b > -Inf;
+        # see page 6, "bounds".
+        if a == Inf
+            return new(prevfloat(a), b)
```

Comment:
There is a case (`cancelminus` I believe) where the subtlety is important. 

In any case, I think you are right. I will change this to throw an error (ArgumentError?) and treat the special case separately.